client/src/ExtensionState.ts

```diff
@@ -119,52 +125,84 @@ export class State {
     return State.client.onReady();
   }
 
-
   // creates a server for the given server binary
-  private static async startServerProcess(serverBin: string): Promise<any> {
-    // test whether java and z3 binaries can be used:
-    Helper.log("Checking Java...");
-    const javaPath = await Helper.getJavaPath();
-    await Helper.spawn(javaPath, ["--version"]);
-    Helper.log("Checking Z3...");
-    const z3Path = Helper.getZ3Path(Helper.isNightly());
-    await Helper.spawn(z3Path, ["--version"]);
+  private static async startServerProcess(serverBin: string): Promise<StreamInfo> {
+    const javaPath = await State.checkDependenciesAndGetJavaPath();
+
+    // spawn Gobra Server and get port number on which it is reachable:
+    const portNr = await new Promise((resolve:(port: number) => void, reject) => {
+      const portRegex = /<GobraServerPort:(\d+)>/;
+      let portFound: boolean = false;
+      function stdOutLineHandler(line: string): void {
+        // check whether `line` contains the port number
+        if (!portFound) {
+          const match = line.match(portRegex);
+          if (match != null && match[1] != null) {
+            const port = Number(match[1]);
+            if (port != NaN) {
+              portFound = true;
+              resolve(port);
+            }
+          }
+        }
+      }
 
-    return new Promise((resolve, reject) => {
-      let server = net.createServer((socket) => {
-        resolve({reader: socket, writer: socket});
-    
-        // event listener for socket reacting to end
-        socket.on('end', () => Helper.log("Disconnected"));
-    
-      }).on('error', (err) => {
-        const msg = `Error while connecting to Gobra server: '${err}'`
+      const processArgs = Helper.getServerProcessArgs(serverBin);
+      Helper.log(`Gobra IDE: Running '${javaPath} ${processArgs.join(' ')}'`);
+      const serverProcess = child_process.spawn(javaPath, processArgs);
+      // redirect stdout to readline which nicely combines and splits lines
+      const rl = readline.createInterface({ input: serverProcess.stdout });
+      rl.on('line', stdOutLineHandler);
+      serverProcess.stdout.on('data', (data) => Helper.logServer(data));
+      serverProcess.stderr.on('data', (data) => Helper.logServer(data));
+      serverProcess.on('close', (code) => {
+        Helper.log(`Gobra Server process has ended with return code ${code}`);
+      });
+      serverProcess.on('error', (err) => {
+        const msg = `Gorba Server process has encountered an error: ${err}`
```

Comment:
is all of this code just to fine the port number?

Reply:
Finding the port number happens as function `stdOutLineHandler`. Lines 150-162 mostly spawn a new process to execute the server binary and forward standard output and error output by the server to a corresponding log